cmd/concourse/web.go

```diff
@@ -117,12 +118,46 @@ func (cmd *WebCommand) populateSharedFlags() error {
 	cmd.RunCommand.Auth.AuthFlags.Clients[cmd.TSACommand.ClientID] = cmd.TSACommand.ClientSecret
 	cmd.RunCommand.Auth.AuthFlags.Clients[cmd.RunCommand.Server.ClientID] = cmd.RunCommand.Server.ClientSecret
 
+	// if we're using the 'aud' as the SystemClaimKey then we want to validate
+	// that the SystemClaimValues contains our TSA Client. If it's not 'aud' then
+	// we can't validate anything
+	if cmd.RunCommand.SystemClaimKey == "aud" {
+
+		// if we're using the default SystemClaimValues then override these values
+		// to make sure they include the TSA ClientID
+		if len(cmd.RunCommand.SystemClaimValues) == 1 {
+			if cmd.RunCommand.SystemClaimValues[0] == "concourse-worker" {
```

Comment:
It doesn't really make sense to do, but in theory, an operator *could* pass in:

```
--tsa-client-id something-else --system-claim-value concourse-worker
```

and it would replace `concourse-worker` with `something-else`. From what I can tell, doing this would basically just be trying to shoot yourself in the foot, so this is probably something we don't care about - just mentioning it anyway.